src/oni2-keyboard-layout/stubs/keyboard-layout-windows.c

```diff
@@ -101,15 +129,16 @@ void characterForNativeCode(
     keyboardState[VK_CONTROL] = 0x80;
   }
 
-  int count = ToUnicodeEx(keycode, scancode, keyboardState, dest, 5, 0, keyboardLayout);
+  int count = keyboardCharacterToUTF8(keycode, scancode, keyboardState, dest, destSize, keyboardLayout);
 
   if (count == -1) {
     keyboardState[VK_SHIFT] = 0x0;
     keyboardState[VK_MENU] = 0x0;
     keyboardState[VK_CONTROL] = 0x0;
 
+    // Doesn't look like this is used?
     UINT spaceKeycode = MapVirtualKeyEx(SPACE_SCAN_CODE, MAPVK_VSC_TO_VK, keyboardLayout);
-    ToUnicodeEx(spaceKeycode, SPACE_SCAN_CODE, keyboardState, dest, 5, 0, keyboardLayout);
+    // ToUnicodeEx(spaceKeycode, SPACE_SCAN_CODE, keyboardState, dest, 5, 0, keyboardLayout);
```

Comment:
@zbaylin - do you happen to know why this `MapVirtualKeyEx`/`ToUnicodeEx` call is needed? It seems like it isn't used based on the `dest[0] = '\0'` below - but wanted to double-check before I removed.

Reply:
I actually don't know -- IIRC I copied that from the atom repo. Your intuition seems correct to me though

Reply:
Ah, cool - thanks @zbaylin !